templates/cluster-template-private.yaml

```diff
@@ -0,0 +1,217 @@
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+metadata:
```

Comment:
When I tried running `make create-workload-cluster` with `CLUSTER_TEMPLATE=cluster-template-private.yaml` I kept getting the following error
```
error: error validating "STDIN": error validating data: [ValidationError(AzureCluster.spec.networkSpec.subnets[0].cidrBlocks): unknown object type "nil" in AzureCluster.spec.networkSpec.subnets[0].cidrBlocks[0], ValidationError(AzureCluster.spec.networkSpec.subnets[1].cidrBlocks): unknown object type "nil" in AzureCluster.spec.networkSpec.subnets[1].cidrBlocks[0], ValidationError(AzureCluster.spec.networkSpec.vnet): missing required field "name" in io.x-k8s.cluster.infrastructure.v1alpha3.AzureCluster.spec.networkSpec.vnet]; if you choose to ignore these errors, turn validation off with --validate=false
make: *** [create-workload-cluster] Error 1
```
even though the regular template worked fine, maybe something with the validation, but didn't find it just yet. Did you try creating the workload cluster this way?

Reply:
That's because you were missing some env vars for subnet CIDRs. I removed all the env vars I added in the template for the test so validation should work now. Keep in mind that your management cluster needs to be in the same vnet for this to work though so it won't work with `make create-workload-cluster` if you're using kind as your management cluster.